addons/mail/models/mail_partner.py

```diff
@@ -0,0 +1,132 @@
+from datetime import datetime
+from odoo import api, fields, models, tools, SUPERUSER_ID
+
+class MailPartnerMixin(models.AbstractModel):
+    """ A mixin to add partner registration support to mail.thread
+        mail_thread model is meant to be inherited (before mail.thread) by any model that needs to
+        add partner registration and auto subscribe of partner on reply.
+        -Register mail informations on "new_message" (email_from,email_cc,partner_id,name)
+        -Create partner on update_message or new_message with respect to the "message_subscription_strategy".
+        -The list of partner to subcribe and/or register are defined by "emails_to_subscribe" 
+        and "partner_temails_to_subscribe". By default, only the "email_from" used for the creation of the record 
+        will be subscribed. 
+        -Once a partner is created, (using odoo interface when posting message or when repplying to a mail), 
+        the field partner_id (can be changed with "message_partner_id_field") is updated with the new partner 
+        corresponding to email_from for every record having the same value for "email_from" and partner field not set 
+        (partner_id or message_partner_id_field if overwritten). 
+    """
+    ON_REPLY_STRATEGY,ON_CREATE_STRATEGY,NEVER_STRATEGY = range(3)
+    _name = 'mail.partner.mixin'
+    
+    email_from = fields.Char('Email', help="Email address of the contact", index=True)
+    email_cc = fields.Text('Global CC', help="These email addresses will be added to the CC field of all inbound and outbound emails for this record before being sent. Separate multiple email addresses with a comma")
+    partner_id = fields.Many2one('res.partner', string='Customer', track_visibility='onchange', index=True, help="Linked partner (optional). Usually created when converting the lead.")
+    name = fields.Char("Subject", translate=True)
```

Comment:
I don't think we should redefine the name field here. It is a common field tha tall model define and I don't think we should hide it in a mixin.  Do some models use another field to store the incoming email subject than name ?

Reply:
I think that the idea here was to keep all the fields related to email information and that can be useful in the mixin. The name is set set when calling message_new in project, crm, and recruitment, when setting the partner_id, email_from, email_cc. The same thing is done in the mixin now and will allow to remove this logic from those models. It will still work if we remove the field declaration in the mixing but it is less clean for me. I can remove it if you think it's better.